src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -523,7 +542,11 @@ impl LogicalScan {
                     })
                     .collect(),
             }
-            .satisfies(required_order)
+            .satisfies(
+                &self
+                    .o2i_col_mapping()
+                    .rewrite_provided_order(required_order),
```

Comment:
```suggestion
                    .rewrite_required_order(required_order),
```
you can read the comments on these two functions and see their slight difference.